Logging Admin V2 crashes when empty input is passed in

```
2022-12-08T19:18:07.504Z DEBUG apiext management/extensions.go:217 handling http request {"method": "CreateOrUpdateOpensearchCluster", "path": "/logging/cluster"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1e4bcbd]
goroutine 429 [running]:
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End.func1()
go.opentelemetry.io/otel/sdk@v1.10.0/trace/span.go:393 +0x2a
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End(0xc00111f380, {0x0, 0x0, 0x0?})
go.opentelemetry.io/otel/sdk@v1.10.0/trace/span.go:432 +0x8ee
panic({0x213f820, 0x3c0d8a0})
runtime/panic.go:884 +0x212
github.com/rancher/opni/plugins/logging/pkg/gateway.(*LoggingManagerV2).generateNodePools(0xc0005502a0, 0xc000c62a20)
github.com/rancher/opni/plugins/logging/pkg/gateway/admin_v2.go:559 +0x3d
github.com/rancher/opni/plugins/logging/pkg/gateway.(*LoggingManagerV2).CreateOrUpdateOpensearchCluster(0xc0005502a0, {0x2883c60?, 0xc001160ed0}, 0xc000c62a20)
github.com/rancher/opni/plugins/logging/pkg/gateway/admin_v2.go:183 +0x27d
github.com/rancher/opni/plugins/logging/pkg/apis/loggingadmin._LoggingAdminV2_CreateOrUpdateOpensearchCluster_Handler.func1({0x2883c60, 0xc001160ed0}, {0x2358cc0?, 0xc000c62a20})
github.com/rancher/opni/plugins/logging/pkg/apis/loggingadmin/loggingadmin_grpc.pb.go:509 +0x78
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1({0x2883c60, 0xc001160d50}, {0x2358cc0, 0xc000c62a20}, 0xc00112c620, 0xc001117578)
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.36.0/interceptor.go:352 +0x724
github.com/rancher/opni/plugins/logging/pkg/apis/loggingadmin._LoggingAdminV2_CreateOrUpdateOpensearchCluster_Handler({0x23a8480?, 0xc0005502a0}, {0x2883c60, 0xc001160d50}, 0xc000c629c0, 0xc00067e210)
github.com/rancher/opni/plugins/logging/pkg/apis/loggingadmin/loggingadmin_grpc.pb.go:511 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00099ddc0, {0x288d6e0, 0xc000bb11e0}, 0xc001126900, 0xc00067ec30, 0x3c21510, 0x0)
google.golang.org/grpc@v1.49.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc00099ddc0, {0x288d6e0, 0xc000bb11e0}, 0xc001126900, 0x0)
google.golang.org/grpc@v1.49.0/server.go:1619 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
google.golang.org/grpc@v1.49.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.49.0/server.go:919 +0x28a
```

--- a/plugins/logging/pkg/errors/errors.go
+++ b/plugins/logging/pkg/errors/errors.go
@@ -64,6 +64,10 @@ func ErrInvalidRetention() error {
 	return fmt.Errorf("duration must be integer and time unit, e.g 7d: %w", ErrInvalidCluster)
 }
 
+func ErrMissingDataNode() error {
+	return fmt.Errorf("request must contain data nodes: %w", ErrInvalidCluster)
+}
+
 func ErrOpensearchRequestFailed(status string) error {
 	return fmt.Errorf("%s: %w", status, ErrOpensearchResponse)
 }

--- a/plugins/logging/pkg/gateway/admin_v2.go
+++ b/plugins/logging/pkg/gateway/admin_v2.go
@@ -162,6 +162,10 @@ func (m *LoggingManagerV2) CreateOrUpdateOpensearchCluster(ctx context.Context, 
 	if !m.validDurationString(lo.FromPtrOr(cluster.DataRetention, "7d")) {
 		return &emptypb.Empty{}, errors.ErrInvalidRetention()
 	}
+	// Input should always have a data nodes field
+	if cluster.GetDataNodes() == nil {
+		return &emptypb.Empty{}, errors.ErrMissingDataNode()
+	}
 	k8sOpensearchCluster := &loggingv1beta1.OpniOpensearch{}
 
 	go m.opensearchManager.SetClient(m.setOpensearchClient)
@@ -553,7 +557,10 @@ func antiAffinityEnabled(pool opsterv1.NodePool) *bool {
 }
 
 func (m *LoggingManagerV2) generateNodePools(cluster *loggingadmin.OpensearchClusterV2) (pools []opsterv1.NodePool, retErr error) {
-	resources, jvm, retErr := generateK8sResources(cluster.DataNodes.MemoryLimit, cluster.DataNodes.CpuResources)
+	resources, jvm, retErr := generateK8sResources(
+		cluster.GetDataNodes().GetMemoryLimit(),
+		cluster.GetDataNodes().GetCpuResources(),
+	)
 	if retErr != nil {
 		return
 	}
@@ -564,18 +571,18 @@ func (m *LoggingManagerV2) generateNodePools(cluster *loggingadmin.OpensearchClu
 		Labels: map[string]string{
 			LabelOpniNodeGroup: "data",
 		},
-		DiskSize:     cluster.DataNodes.DiskSize,
-		NodeSelector: cluster.DataNodes.NodeSelector,
+		DiskSize:     cluster.GetDataNodes().GetDiskSize(),
+		NodeSelector: cluster.GetDataNodes().GetNodeSelector(),
 		Resources:    resources,
 		Jvm:          jvm,
-		Tolerations:  convertTolerations(cluster.DataNodes.Tolerations),
+		Tolerations:  convertTolerations(cluster.GetDataNodes().GetTolerations()),
 		Affinity: func() *corev1.Affinity {
-			if lo.FromPtrOr(cluster.DataNodes.EnableAntiAffinity, false) {
+			if lo.FromPtrOr(cluster.GetDataNodes().EnableAntiAffinity, false) {
 				return m.generateAntiAffinity("data")
 			}
 			return nil
 		}(),
-		Persistence: convertPersistence(cluster.DataNodes.Persistence),
+		Persistence: convertPersistence(cluster.GetDataNodes().GetPersistence()),
 	}
 
 	var extraControlPlanePool, splitPool bool

